
Here show the scripts I wrote during my study.

## Perl
```
"get_accession_strain-name_location_length_collection_add_vp1_version3_final.pl"
```
This is a __perl__ script adopted Bioperl to parse the meta data retrieved from GenBank 
nucleotide database, the downloaded files were in .gb format,using this script, the VP1 
nucleotide sequences of around 2,000 enterovirus d68 were extracted from the .gb file.
Thus, the download analysis such as alignment, phylogenetic, mutation rate calculation could 
be carried out, gigantic amount of manpower was saved, without this tool, this data cleaning 
work would be quite labor intensive.

## R
```
[transform_data_add_Yr_Month_then_monthly_plot.R]()
```

## Data visualization ability



## MySQL
```
"age_group_proportion.sql"
```
In order to manage over 2 Million rows clinical records data during my MPhil study,a __MySQL database__
was established. The reason for using MySQL was that the 2 Million row data would be a big challenge
for management and analysis if EXCEL was adopted. These two scripts were used for summarizing the 
demographics for clinical samples. The data was stratified by age and gender. The SQL scripts were used
for counting the number of patients assigned to diffent groups.

##



