# data_analyze
here show the scripts I wrote during my study.

# get_accession_strain-name_location_length_collection_add_vp1_version3_final.pl

This is a perl script adopted Bioperl to parse the meta data retrieved from GenBank 
nucleotide database, the downloaded files were in .gb format,using this script, the VP1 
nucleotide sequences of around 2,000 enterovirus d68 were extracted from the .gb file.
Thus, the download analysis such as alignment, phylogenetic, mutation rate calculation could 
be carried out, gigantic amount of manpower was saved, without this tool, this data cleaning 
work would be quite labor intensive.

#


